rdt/transformers/id.py

```diff
@@ -164,27 +203,35 @@ def _fit(self, data):
         self.reset_randomization()
         self.data_length = len(data)
 
+        if self.cardinality_rule == 'match':
+            is_nan = int(pd.isna(data).any())  # nans count as a unique value
+            self._data_cardinality = data.nunique() + is_nan
```

Comment:
minor: I think you can just do 
```python
self._data_cardinality = data.nunique(dropna=False)
```

Reply:
@amontanez24 It unfortunately doesn't work, @pvk-developer mentioned a similar [idea](https://github.com/sdv-dev/RDT/pull/971#discussion_r2024614878).

```python3
>>> pd.Series(['1', np.nan, None]).nunique(dropna=False)
3
```